app/controllers/concerns/account_concern.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module AccountConcern
+  extend ActiveSupport::Concern
+
+  ACCOUNT_SESSION_REQUEST_HEADER_NAME = "HTTP_GOVUK_ACCOUNT_SESSION"
+  ACCOUNT_SESSION_RESPONSE_HEADER_NAME = "GOVUK-Account-Session"
+  ACCOUNT_END_SESSION_HEADER_NAME = "HTTP_GOVUK_ACCOUNT_END_SESSION"
+  ACCOUNT_SESSION_DEV_COOKIE_NAME = "govuk_account_session"
+
+  included do
+    before_action :fetch_account_session_header
+    before_action :set_account_session_header
+    helper_method :logged_in?
+
+    attr_accessor :account_session_header
+  end
+
+  def logged_in?
+    account_session_header.present?
+  end
+
+  def fetch_account_session_header
+    @account_session_header =
+      if request.headers.to_h[ACCOUNT_SESSION_REQUEST_HEADER_NAME]
+        request.headers.to_h[ACCOUNT_SESSION_REQUEST_HEADER_NAME]
```

Comment:
Don't need the `to_h` any more

Reply:
done here: https://github.com/alphagov/frontend/pull/2668/commits/205008d03ba25a5f727f330c48247c0ccd055d13